src/runtime/formal/ops/ops.cc

```diff
@@ -682,34 +622,50 @@ static void take(CVMArgValue x,
 
     if(axis < 0) { axis += xndim; }
 
-    std::vector<size_t> x_shape_size(xndim, 1), indices_shape_size(indices_ndim, 1);
-    for (int i = xndim-2; i >= 0; --i) {
-      x_shape_size[i] = x_shape_size[i+1] * X_shape[i+1];
-    }
-    for (int i = indices_ndim-2; i >= 0; --i) {
-      indices_shape_size[i] = indices_shape_size[i+1] * Indices_shape[i+1];
+    for (Indices yIdx(Y_shape); !yIdx.End(); yIdx++) {
+      Indices xIdx(X_shape), idxIdx(Indices_shape);
```

Comment:
Will move the `Indices` instance out of the loop be better, since indices flatten index calculation
will use cache.

Do not forget to reset the Indices at the beginning of each loop.

Reply:
`xIdx`, `idxIdx` are modified in the for-loop, cache will not work